### DISCUSSION TITLE
Finding the installation/execution directory of Theia based IDE

### DISCUSSION THREAD

**Discussion by dagoston93** [eclipse-theia/theia#13602] [Q&A] [answered]
Hi,

I am exploring Theia IDE for my company to decide whether if it is suitable for our purpose.
I would need to bundle some executables with the IDE, that would be executed during our workflow.
I also would like to add a command that opens the user manual, that would be bundled with the app as well.

But I cannot find an API that retrives the install path of Theia.
I tried to modify the 'product' extension in the **Theia IDE**, to try to open the User Manual markdown file, but with little success.
If I try to use __dirname, then I get a '\' and if I read the directory as follows.
`var e = await (await this.remoteFileSystemProvider.readdir(uri)).concat();`
I get the contents of my C drive. (If I copy my file there, then **OpenService** opens it fine...)
If I try:
`var instDir = process.execPath;`
I get an error that process is undefined.
(I added these changes to **theia-extensions/product/src/browser/theia-ide-contribution.tsx**)

I am pretty sure, that many people have to run custom command line tools in a background or open bundled files, so I must somehow missed where it is described how to do it.

Any help is greatly appreciated.

BR,
Agoston

**Comment by msujew** [comment, marked as answer]
Hey @dagoston93,

it looks like you're attempting to use Node.js API in browser code. Variables like `__dirname` get a polyfill in the browser, while other objects such as `process` only exist in Node.js (the backend). 

In order to get the installation directory in the frontend, you can use the [`ApplicationServer#getApplicationRoot`](https://github.com/eclipse-theia/theia/blob/fd5be308a0da5c1e9f660c3afa86ebe60fc5bf8e/packages/core/src/common/application-protocol.ts#L26) method.

**Reply by dagoston93** [reply]
Thanks a lot, this is exactly what I needed :)
